custom_components/isy994/__init__.py

```diff
@@ -575,6 +647,40 @@ def on_control(self, event: NodeProperty) -> None:
 
         self.hass.bus.fire("isy994_control", event_data)
 
+    @property
+    def device_info(self):
+        """Return the device_info of the device."""
+        if hasattr(self._node, "protocol") and self._node.protocol == PROTO_GROUP:
+            # not a device
+            return None
+
+        device_info = {"name": self.name}
+        if hasattr(self._node, "primary_node"):
+            device_info["identifiers"] = {(DOMAIN, self._node.primary_node)}
+        if hasattr(self._node, "protocol"):
+            device_info["manufacturer"] = self._node.protocol
+        # ISYv5 Device Types
+        if hasattr(self._node, "node_def_id") and self._node.node_def_id is not None:
+            device_info["model"] = self._node.node_def_id
+            # Numerical Device Type
+            if hasattr(self._node, "type") and self._node.type is not None:
+                device_info["model"] += f" {self._node.type}"
+            # Z-Wave Device Type Category
+            if (
+                hasattr(self._node, "devtype_cat")
+                and self._node.devtype_cat is not None
+            ):
+                device_info["model"] += f" {self._node.devtype_cat}"
+        device_info["via_device"] = (DOMAIN, self._node.isy.configuration["uuid"])
+        # PyISY does not expose the software version of the device
```

Comment:
Let's go ahead and expose the ISY firmware as the `sw_version` (`self._node.isy.configuration["firmware"]`). I looked at a few other implementations and if the device sw version isn't exposed, the hub's is.

Reply:
https://developers.home-assistant.io/docs/device_registry_index/
The docs say its supposed to be the `The firmware version of the device.`

That works for the isy itself.  I'm going to leave it out for now since we can always add it later once get it the version pulling from the device